src/main/kotlin/uk/gov/justice/digital/hmpps/courtregister/jpa/BuildingRepository.kt

```diff
@@ -19,27 +21,31 @@ import javax.persistence.OneToMany
 interface BuildingRepository : CrudRepository<Building, Long>
 
 @Entity
+@EntityListeners(AuditingEntityListener::class)
 data class Building(
   @Id
   @GeneratedValue(strategy = GenerationType.IDENTITY)
   val id: Long? = null,
+
   @ManyToOne(optional = false, fetch = FetchType.LAZY)
   @JoinColumn(name = "COURT_CODE")
-  val court: Court,
-  val subCode: String?,
-  val buildingName: String?,
-  val street: String?,
-  val locality: String?,
-  val town: String?,
-  val county: String?,
-  val postcode: String?,
-  val country: String?,
+  var court: Court,
+
+  var subCode: String?,
+  var buildingName: String?,
+  var street: String?,
+  var locality: String?,
+  var town: String?,
+  var county: String?,
+  var postcode: String?,
+  var country: String?,
+
   @CreatedDate
-  var createdDatetime: LocalDateTime = LocalDateTime.now(),
+  var createdDatetime: LocalDateTime = LocalDateTime.MIN,
   @LastModifiedDate
-  var lastUpdatedDatetime: LocalDateTime = LocalDateTime.now(),
+  var lastUpdatedDatetime: LocalDateTime = LocalDateTime.MIN,
 
   @OneToMany(cascade = [CascadeType.ALL], mappedBy = "building")
-  var contacts: List<Contact> = listOf()
+  val contacts: MutableList<Contact>? = mutableListOf()
```

Comment:
I wonder if we could have a getter that returns a non-mutable list?  But then saying that I suspect behind the scenes it's just getting the mutable list and adding to them so maybe not :man_shrugging: